### PR TITLE
feat(kms): allow specifying key spec and key usage

### LINKED ISSUE #5639: aws-kms - Asymmetric keys
Please let me know if I missed something, is there not a way to create an asymmetric key via the aws-kms module?   I've been grepping and googling and can't find anything anywhere..

### PR DESCRIPTION
**Pull request by dacci**
This allows specifying key spec and key usage, so you can create asymmetric keys.

closes #5639

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 1]
```diff
@@ -282,6 +361,20 @@ export interface KeyProps {
    */
   readonly enabled?: boolean;
 
+  /**
+   * The cryptographic configuration of the key. The valid value depends on usage of the key.
+   *
+   * @default - KeySpec.SYMMETRIC_DEFAULT
```
Comment (njlynch) [inline packages/@aws-cdk/aws-kms/lib/key.ts:367]:
```suggestion
   * @default KeySpec.SYMMETRIC_DEFAULT
```

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 2]
```diff
@@ -282,6 +361,20 @@ export interface KeyProps {
...
+   */
+  readonly keySpec?: KeySpec;
+
+  /**
+   * The cryptographic operations for which the key can be used.
+   *
+   * @default - KeyUsage.ENCRYPT_DECRYPT
```
Comment (njlynch) [inline packages/@aws-cdk/aws-kms/lib/key.ts:374]:
```suggestion
   * @default KeyUsage.ENCRYPT_DECRYPT
```

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 3]
```diff
@@ -282,6 +361,20 @@ export interface KeyProps {
    */
   readonly enabled?: boolean;
 
+  /**
+   * The cryptographic configuration of the key. The valid value depends on usage of the key.
```
Comment (njlynch) [inline packages/@aws-cdk/aws-kms/lib/key.ts:365]: I think it's important to capture the same warning present in the docs here and for `keyUsage`:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html#cfn-kms-key-keyusage

Effectively a nice big "**WARNING** - Changing this property will..." in the docstring so it's clear that users can't arbitrarily change these.

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 4]
```diff
@@ -428,6 +542,8 @@ export class Key extends KeyBase {
       description: props.description,
       enableKeyRotation: props.enableKeyRotation,
       enabled: props.enabled,
+      keySpec: props.keySpec,
+      keyUsage: props.keyUsage,
```
Comment (njlynch) [inline packages/@aws-cdk/aws-kms/lib/key.ts:546]: One thing to consider here: as-is, the implementation will mean that every customer with a key today will see an update for each of their keys when the upgrade the CDK, with both of these properties created. Given that these are the default values anyway, that _should_ be ok, just potentially mildly annoying.

The alternative -- which is more backwards-compatible -- would be to leave these as undefined if the customer does not specify them. I think I have a slight preference for that.

**Review by njlynch — DISMISSED** [review]
Looks great! Thanks for the contribution!

Can you please also include a section in the README with a brief description of the feature and an example? Doesn't have to be long, and feel free to steal phrasing from the official docs.

File: packages/@aws-cdk/aws-kms/lib/key.ts (hunk 4 above)
Comment (dacci) [reply packages/@aws-cdk/aws-kms/lib/key.ts:552]: I agree with you. So, I copied them to local variables for validation to keep the `props` unchanged, and existing tests passed without change. Should I implement in different way?

https://github.com/dacci/aws-cdk/blob/98231949897402007317c2b5ed89dce1514587e4/packages/%40aws-cdk/aws-kms/lib/key.ts#L507-L508

File: packages/@aws-cdk/aws-kms/lib/key.ts (hunk 3 above)
Comment (dacci) [reply packages/@aws-cdk/aws-kms/lib/key.ts:365]: As I couldn't find docstrings of constructs using WARNING, I mentioned this with IMPORTANT.

Comment (njlynch) [reply packages/@aws-cdk/aws-kms/lib/key.ts:365]: Sure, that works. Thanks!

File: packages/@aws-cdk/aws-kms/lib/key.ts (hunk 4 above)
Comment (njlynch) [reply packages/@aws-cdk/aws-kms/lib/key.ts:552]: Nope -- this is perfect. I must have mis-read it the first time. Sorry about that. 🤦‍♂️

**Review by njlynch — APPROVED** [review]
Thanks again for the contribution!

**Pull request merged**